src/library/scala/concurrent/Future.scala

```diff
@@ -844,34 +843,7 @@ object Future {
   final def traverse[A, B, M[X] <: IterableOnce[X]](in: M[A])(fn: A => Future[B])(implicit bf: BuildFrom[M[A], B, M[B]], executor: ExecutionContext): Future[M[B]] =
     in.iterator.foldLeft(successful(bf.newBuilder(in))) {
       (fr, a) => fr.zipWith(fn(a))(Future.addToBuilderFun)
-    }.map(_.result())(if (executor.isInstanceOf[BatchingExecutor]) executor else InternalCallbackExecutor)
-
-
-  // This is used to run callbacks which are internal
-  // to scala.concurrent; our own callbacks are only
-  // ever used to eventually run another callback,
-  // and that other callback will have its own
-  // executor because all callbacks come with
-  // an executor. Our own callbacks never block
-  // and have no "expected" exceptions.
-  // As a result, this executor can do nothing;
-  // some other executor will always come after
-  // it (and sometimes one will be before it),
-  // and those will be performing the "real"
-  // dispatch to code outside scala.concurrent.
-  // Because this exists, ExecutionContext.defaultExecutionContext
-  // isn't instantiated by Future internals, so
-  // if some code for some reason wants to avoid
-  // ever starting up the default context, it can do so
-  // by just not ever using it itself. scala.concurrent
-  // doesn't need to create defaultExecutionContext as
-  // a side effect.
-  private[concurrent] object InternalCallbackExecutor extends ExecutionContextExecutor with BatchingExecutor {
-    override final def submitForExecution(runnable: Runnable): Unit = runnable.run()
-    final override def execute(runnable: Runnable): Unit = submitSyncBatched(runnable)
-    override final def reportFailure(t: Throwable): Unit =
-      ExecutionContext.defaultReporter(new IllegalStateException("problem in scala.concurrent internal callback", t))
```

Comment:
I'm just noticing that this behaviour changes slightly - internal exceptions are no longer wrapped in `IllegalStateException`s - is that desired? (I realise we don't want to duplicate logic)

Reply:
@NthPortal The reason I removed it is because it is not related to scala.concurrent internal callbacks anymore, and the stacktrace will most likely be non-deterministic to look at. :/